extensions/2.0/Khronos/KHR_materials_ior/README.md

```diff
@@ -0,0 +1,144 @@
+# KHR\_materials\_ior
+
+## Contributors
+
+* Tobias Haeussler, Dassault Systemes [@proog128](https://github.com/proog128)
+* Bastian Sdorra, Dassault Systemes [@bsdorra](https://github.com/bsdorra)
+* Ben Houston, ThreeKit [@bhouston](https://github.com/bhouston)
+* Richard Sahlin, IKEA [@rsahlin](https://github.com/rsahlin)
+* Norbert Nopper, UX3D [@UX3DGpuSoftware](https://twitter.com/UX3DGpuSoftware)
+* Ed Mackey, AGI [@emackey](https://twitter.com/emackey)
+
+Copyright (C) 2018-2020 The Khronos Group Inc. All Rights Reserved. glTF is a trademark of The Khronos Group Inc.
+See [Appendix](#appendix-full-khronos-copyright-statement) for full Khronos Copyright Statement.
+
+## Status
+
+Experimental
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+The dielectric BRDF of the metallic-roughness material in glTF uses a fixed value of 1.5 for the index of refraction. This is a good fit for many plastics and glass, but not for other materials like water or asphalt, sapphire or diamond. This extensions allows users to set the index of refraction to a certain value.
+
+## Extending Materials
+
+The index of refraction of a material is configured by adding the `KHR_materials_ior` extension to any glTF material. 
+
+```json
+{
+    "materials": [
+        {
+            "extensions": {
+                "KHR_materials_ior": {
+                    "ior": 1.4,
+                }
+            }
+        }
+    ]
+}
+```
+
+| |Type|Description|Required|
+|-|----|-----------|--------|
+| **ior** | `number` | The index of refraction. | No, default: `1.5`|
+
+Typical values for the index of refraction range from 1 to 2. In rare cases values greater than 2 are possible. The following table shows some materials and their index of refraction:
+
+| Material     | Index of Refraction |
+|--------------|---------------------|
+| Air          | 1.0                 |
+| Water        | 1.33                |
+| Eyes         | 1.38                |
+| Window Glass | 1.52                |
+| Sapphire     | 1.76                |
+| Diamond      | 2.42                |
+
+The reflectance at normal incidence (F0) of dielectric materials is computed from the IOR as follows:
+
+```
+f0 = ((ior - outside_ior) / (ior + outside_ior))^2
+```
+
+The `outside_ior` determines the index of refraction of the medium on the outside. If the renderer does not track the IOR when traversing nested dielectrics, it can assume `outside_ior = 1`.
+
+For `ior = 1.5` and `outside_ior = 1`, the result is `f0 = 0.04`. This is the fixed value used in the glTF 2.0 metallic-roughness material. The extension makes this value configurable.
+
+
+## Implementation
+
+*This section is non-normative*
+
+The extension changes the computation of the Fresnel term defined in [Appendix B](/specification/2.0/README.md#appendix-b-brdf-implementation) to the following:
+
+```
+dielectricSpecular = ((ior - 1)/(ior + 1))^2
```

Comment:
Perhaps worth to mention that implementations are free to keep track of IOR for objects/materials that reside inside some other medium than air?
Thus replacing the 1 with the ior for that medium.
One such example could be a (submerged) submarine with a glass window.

Reply:
Specifically, when the camera itself resides in a non-air medium, such as water.  I think this is the same concern that my comment was struggling to reach.

Reply:
This is mentioned in the "Extending Materials" section that describes how the parameters behave:

> The `outside_ior` determines the index of refraction of the medium on the outside. If the renderer does not track the IOR when traversing nested dielectrics, it can assume `outside_ior = 1`.

It is also mentioned in KHR_materials_volume (#1726), because it is relevant only for volumetric objects:

> Incident and transmitted index of refraction have to be correctly set by the renderer, depending on whether light enters or leaves the object. An algorithm for tracking the IOR through overlapping objects is described in [Schmidt C., Budge B. (2002): Simple Nested Dielectrics in Ray Traced Images](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.125.5212&rep=rep1&type=pdf).

I didn't mention it in the "Implementation" section because so far I tried to keep this section as simple as possible, focused on real-time implementations and with little interactions between extensions as possible (i.e., as the default glTF 2.0 metallic-roughness material is thin-walled, we cannot reside in another medium than air; except if the camera is placed in something else, but that seems to be a rather rare use-case).